### ISSUE TITLE
Does Dataform support `ALLOW_FIELD_ADDITION` for incremental table related jobs ?

### ISSUE THREAD

**Issue by Rstar1998** [dataform-co/dataform#1385]
Big Query does support addition of new column if it does not exist. For incremental tables there might be instances that a new column might be added. Hence it would be beneficial to have a parameter like ALLOW_FIELD_ADDITION=true   which will add new columns in the target table if they are added in the source table

**Comment by nmohseny-atz** [comment]
+1

**Comment by astorrs** [comment]
Yes please! This is very painful in models with heavy use of incrementals.

**Comment by Ekrekr** [comment]
Note: this implementation for this is similar to the request in https://github.com/dataform-co/dataform/issues/1359, as it interacts with the API in the same way.
